### PR TITLE
Add runner for seviri

### PR DESCRIPTION
**Pull request by smhi-erik**
For seviri the orbit number is not enough for finding files since seviri do not have any orbit number. Therefore added the start time as well

### CODE REVIEW

**Comment by mraspaud** [comment]
could you run flake8 with the docstring plugin on your PR ? I see for example some trailing spaces that should be removed

File: nwcsafpps_runner/publish_and_listen.py [hunk 1]
```diff
@@ -29,6 +29,13 @@
 from nwcsafpps_runner.utils import (SUPPORTED_PPS_SATELLITES,
                                     SUPPORTED_METEOSAT_SATELLITES)
 
+
+#===============================================================================
+# from utils import (SUPPORTED_PPS_SATELLITES,
+#                                     SUPPORTED_METEOSAT_SATELLITES)
+#===============================================================================
+
+
```
Comment (mraspaud) [inline nwcsafpps_runner/publish_and_listen.py:38]: Is this necessary?

File: nwcsafpps_runner/utils.py [hunk 2]
```diff
@@ -496,22 +496,21 @@ def get_pps_inputfile(platform_name, ppsfiles):
...
-    picking up an older scene we check the file modifcation time, and if the
-    file is too old we discard it!
-
+def get_outputfiles(path, platform_name, orb, st_time, **kwargs):
+    """From the directory path and satellite id and orbit number and start time,
+    scan the directory and find all pps output files matching that scene and
+    return the full filenames. Since the orbit number is unstable there might be
+    more than one scene with the same orbit number and platform name. In order
+    to avoid picking up an older scene we check the file modifcation time, and
+    if the file is too old we discard it!
```
Comment (mraspaud) [inline nwcsafpps_runner/utils.py:505]: Did you install the flake8-docstring plugin ? the docstyle in python says there should be one summary line and then the description after a blank line.

File: nwcsafpps_runner/pps2018_runner.py [hunk 3]
```diff
@@ -47,17 +47,19 @@
...
 # import pdb
 # from config import get_config  # @UnresolvedImport @UnusedImport
 # from config import MODE  # @UnresolvedImport @Reimport
 # from config import CONFIG_FILE  # @UnresolvedImport @UnusedImport
 # from config import CONFIG_PATH  # @UnresolvedImport @UnusedImport
-# 
+#
 # from utils import ready2run, publish_pps_files  # @UnresolvedImport @UnusedImport
 # from utils import (get_sceneid, prepare_pps_arguments,              # @UnresolvedImport @UnusedImport
 #                    create_pps2018_call_command, get_pps_inputfile,  # @UnresolvedImport @UnusedImport
```
Comment (mraspaud) [inline nwcsafpps_runner/pps2018_runner.py:65]: Can these be removed or do they serve a purpose ?

**Comment by mraspaud** [comment]
Also: could a test be added to check the new feature?

File: nwcsafpps_runner/pps2018_runner.py (hunk 3 above)
Comment (smhi-erik) [reply nwcsafpps_runner/pps2018_runner.py:65]: They do serve a big purpose for me. But if they are in the way I will figure out a different way to keep them.

File: nwcsafpps_runner/utils.py (hunk 2 above)
Comment (smhi-erik) [reply nwcsafpps_runner/utils.py:505]: Yes, but I did not go into details in functions that I did not create or spent a lot of time in.

**Comment by mraspaud** [comment]
@smhi-erik don't close the PR, just push to your branch for updates!

File: nwcsafpps_runner/utils.py (hunk 2 above)
Comment (smhi-erik) [reply nwcsafpps_runner/utils.py]: Uppdated in new PR

File: nwcsafpps_runner/publish_and_listen.py (hunk 1 above)
Comment (smhi-erik) [reply nwcsafpps_runner/publish_and_listen.py]: Removed in new PR

File: nwcsafpps_runner/pps2018_runner.py (hunk 3 above)
Comment (smhi-erik) [reply nwcsafpps_runner/pps2018_runner.py]: Removed in new PR

File: nwcsafpps_runner/utils.py [hunk 4]
```diff
@@ -496,16 +496,20 @@ def get_pps_inputfile(platform_name, ppsfiles):
...
     scan the directory and find all pps output files matching that scene and
     return the full filenames. Since the orbit number is unstable there might be
     more than one scene with the same orbit number and platform name. In order
     to avoid picking up an older scene we check the file modifcation time, and
-    if the file is too old we discard it!
+    if the file is too old we discard it! For a more specific search patern the
+    start time can be used, just add st_time=start-time
     """
-
+
```
Comment (mraspaud) [inline nwcsafpps_runner/utils.py:510]: trailing spaces...

File: nwcsafpps_runner/tests/test_utils.py [hunk 5]
```diff
@@ -21,24 +21,79 @@
...
+        #: These files should not be found although the start time is correct
+        f3 = d / "S_NWC_CMAPROB_noaa15_54321_19810305T0715000Z_19810305T0730000Z.{}".format(typ)
+        f3.write_text("wrong orbit and correct time")
+        #: TODO: Create testfile older than 90 min but with correct orbitnumber
+
+    #: Test xml files without start time
+    typ = "xml"
+    create_files(d, typ)
+    expected = ["{}/S_NWC_CMAPROB_noaa15_12345_19810305T0715000Z_19810305T0730000Z.{}".format(d.__str__(), typ),
+                "{}/S_NWC_CMAPROB_noaa15_12345_19810305T0745000Z_19810305T0800000Z.{}".format(d.__str__(), typ)]
```
Comment (mraspaud) [inline nwcsafpps_runner/tests/test_utils.py:59]: Use 'os.path.join' instead of `/`, for portability

File: nwcsafpps_runner/tests/test_utils.py [hunk 6]
```diff
@@ -21,24 +21,79 @@
...
+    and is so disregard it. This behaviour can't be tested at the moment.
+    To do so either one file needs to be created more than 90 mins ago
+    or the os.stat should be modified so the so.stat thinks the file was
+     created more than 90 mins ago. The file should than not be found.
+    """
+    #: Create temp_path
+    d = tmp_path / "export"
+    d.mkdir()
+    #: Print tmp_path
+    print(d.__str__())
```
Comment (mraspaud) [inline nwcsafpps_runner/tests/test_utils.py:40]: Please remove the print.

File: nwcsafpps_runner/tests/test_utils.py [hunk 7]
```diff
@@ -21,24 +21,79 @@
...
-import unittest
 
 
-class Test(unittest.TestCase):
+def test_outputfiles(tmp_path):
+    """Test get_outputfiles.
+
+    get_outputfiles uses os.stat to test if a file is older than 90 min,
+    and is so disregard it. This behaviour can't be tested at the moment.
+    To do so either one file needs to be created more than 90 mins ago
```
Comment (mraspaud) [inline nwcsafpps_runner/tests/test_utils.py:32]:
```suggestion
    TODO: either one file needs to be created more than 90 mins ago
```

File: nwcsafpps_runner/tests/test_utils.py [hunk 8]
```diff
@@ -50,46 +49,45 @@ def create_files(d, typ):
...
         f3.write_text("wrong orbit and correct time")
-        #: TODO: Create testfile older than 90 min but with correct orbitnumber
 
     #: Test xml files without start time
     typ = "xml"
     create_files(d, typ)
-    expected = ["{}/S_NWC_CMAPROB_noaa15_12345_19810305T0715000Z_19810305T0730000Z.{}".format(d.__str__(), typ),
-                "{}/S_NWC_CMAPROB_noaa15_12345_19810305T0745000Z_19810305T0800000Z.{}".format(d.__str__(), typ)]
+    expected = [os.path.join(d.__str__(), "S_NWC_CMAPROB_noaa15_12345_19810305T0715000Z_19810305T0730000Z.{}".format(typ)),
+                os.path.join(d.__str__(), "S_NWC_CMAPROB_noaa15_12345_19810305T0745000Z_19810305T0800000Z.{}".format(typ))]
```
Comment (mraspaud) [inline nwcsafpps_runner/tests/test_utils.py]: ok, if you have to do `__str__` I suppose `d` is a pathlib object ? in which case is more natural to use the `/` operator:
```suggestion
    expected = [d / "S_NWC_CMAPROB_noaa15_12345_19810305T0715000Z_19810305T0730000Z.{}".format(typ)),
                d / "S_NWC_CMAPROB_noaa15_12345_19810305T0745000Z_19810305T0800000Z.{}".format(typ))]
```

Comment (smhi-erik) [reply nwcsafpps_runner/tests/test_utils.py]: Now I just did  
`os.path.join(d, "S_NWC_CMAPROB_noaa15_12345_19810305T0745000Z_19810305T0800000Z.{}".format(typ))]`
instead. Bt maybe `/` is better?

Comment (mraspaud) [reply nwcsafpps_runner/tests/test_utils.py]: It's fine as it is!

**Comment by adybbroe** [comment]
I still don't manage to run pytest:
```
========================================================================= ERRORS ==========================================================================
__________________________________________________ ERROR collecting nwcsafpps_runner/tests/test_utils.py __________________________________________________
ImportError while importing test module '/home/a000680/usr/src/forks/pytroll-pps-runner/nwcsafpps_runner/tests/test_utils.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
nwcsafpps_runner/tests/test_utils.py:24: in <module>
    from nwcsafpps_runner.utils import get_outputfiles
E   ModuleNotFoundError: No module named 'nwcsafpps_runner'
================================================================= short test summary info =================================================================
ERROR nwcsafpps_runner/tests/test_utils.py
```

**Pull request merged**